### DISCUSSION TITLE
Change localhost connection

### DISCUSSION THREAD

**Discussion by cesar-munuera** [DSD-DBS/capella-collab-manager#1658] [Q&A] [answered]
Hello everyone.

I have recently downloaded and installed Capella Collab (v3.2.9), and it would be so useful to connect this service from another pc of the same LAN.
Now it is not possible to reach service from outside (same LAN), because it is running at localhost. But localhost it is NOT the IP address from the PC which is running Capella Collab.

Is there any kind of configuration, that expose Capella Collab at the external IP?

Thanks!

**Comment by MoritzWeber0** [comment]
Hi @cesar-mp-atica,

which software do you use to run the Kubernetes cluster? `k3d`? 
In addition, can share the error message that you get in the browser when connecting via the IP address of the machine? 

Thanks!

**Reply by cesar-munuera** [reply]
Hi @MoritzWeber0 !

Yes, we are using k3d. We have followed the README v3.2.9 steps: Download repo -> make create-cluster -> (point 2) export CAPELLA_VERSIONS="6.0.0 6.1.0" -> DEVELOPMENT_MODE=1 make deploy.

It shows a 404 error. For example, lets suppose that the IP address of the PC that runs Capella Collab is 10.42.0.189. So If I try to connect, in the same LAN, with other PC (10.42.0.190), in this client pc shows a 404 error.

I can only use Capella Collab, if I access via localhost, in the host pc.

Thanks in advance.

**Reply by MoritzWeber0** [reply, marked as answer]
Try to update the following values in the `values.yaml`:

- Set `general.wildcardHost` to `True` (this will disable strict hostname checking)
- Replace `localhost` in `backend.authentication.oauth.endpoints.authorization` and `backend.authentication.oauth.redirectURI` with the external IP address of the node where the Capella Collaboration Manager runs on (in your example 10.42.0.189).

Then rerun the `DEVELOPMENT_MODE=1 make deploy` target. 

Let me know if that helps.

**Reply by cesar-munuera** [reply]
Hello.

The first point at `values.yaml`, the `general.wildcardHost` to `False`, is already by default. Can [see here.](https://github.com/DSD-DBS/capella-collab-manager/blob/v3.2.9/helm/values.yaml)
Unfortunatelly, this step does not work. 

However, I have set the `general.wildcardHost` to `True`. Also, the `general.host` to my IP address, and the same second point you suggested (backend.authentication.oauth...). And it works!

This method is a bit of a hack since it hardcodes the IP address. So is there any way to get my IP dynamically?

Thanks in advance!

**Reply by MoritzWeber0** [reply]
Ah, thanks for the correction. `general.wildcardHost` has to be True 👍 

It's currently not possible to derive the redirectURI from the current IP address. The usual deployment method uses a fixed hostname instead of an IP address.  

You could either assign your host machine a static IP address or assign it a hostname which can be resolved by your DNS server. 

One simple example would be to use `capella-collab.external` as hostname, which you'd use instead of `localhost` in the `backend.authentication.oauth.[...]` configuration. To make your client understand the hostname, add to the `/etc/hosts` file on the client device:

```zsh
<IP-of-CCM-server> capella-collab.external
``` 

After that, your browser will be able to resolve `capella-collab.external` and the CCM will use `capella-collab.external` in all redirect URIs.

**Reply by cesar-munuera** [reply]
Okey, thanks.

We will try this in the future. Thanks!!
